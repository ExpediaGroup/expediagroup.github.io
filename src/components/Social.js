/*
Copyright 2021 Expedia, Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React from "react";
import styles from "./Social.module.css";
import TwitterTimeline from "./TwitterTimeline";
import BlogPosts from "./BlogPosts";
import posts from '@site/static/posts.json';
import SocialHeader from "./SocialHeader";
import SocialFooter from "./SocialFooter";
import careersImageUrl from '@site/static/img/careers.jpg';

function Social({socialConfig}) {
    return (
        <section className={styles.social}>
            <div className="container">
                <div className="row">
                    <div className='col col--4'>
                        <div className={styles.socialColumn}>
                            <div className={styles.socialHeaderAndBody}>
                                <SocialHeader title={socialConfig.blog.title}/>
                                <div className={styles.socialBody}>
                                    <BlogPosts posts={posts}/>
                                </div>
                            </div>
                            <SocialFooter text={socialConfig.blog.footerText} link={socialConfig.blog.link}/>
                        </div>
                    </div>
                    <div className='col col--4'>
                        <div className={styles.socialColumn}>
                            <div className={styles.socialHeaderAndBody}>
                                <SocialHeader title={socialConfig.careers.title}/>
                                <div className={styles.socialBody}>
                                    <a href={socialConfig.careers.link} target="_blank">
                                        <img className={styles.careersImage} src={careersImageUrl} alt={socialConfig.careers.title}/>
                                    </a>
                                </div>
                            </div>
                            <SocialFooter text={socialConfig.careers.footerText} link={socialConfig.careers.link}/>
                        </div>
                    </div>
                    <div className='col col--4'>
                        <div className={styles.socialColumn}>
                            <div className={styles.socialHeaderAndBody}>
                                <SocialHeader title={socialConfig.twitter.title}/>
                                <div className={styles.socialBody}>
                                    <TwitterTimeline twitterLink={socialConfig.twitter.link}/>
                                </div>
                            </div>
                            <SocialFooter text={socialConfig.twitter.footerText} link={socialConfig.twitter.link}/>
                        </div>
                    </div>
                </div>
            </div>
        </section>
    )
}

export default Social;
