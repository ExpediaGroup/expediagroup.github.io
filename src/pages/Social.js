import React from "react";
import styles from "./Social.module.css";
import clsx from "clsx";
import TwitterTimeline from "./TwitterTimeline";
import ArrowLink from "./ArrowLink";

function Social({socialLinks}) {
    return (
        <section className={styles.social}>
            <div className="container">
                <div className="row">
                    <div className={clsx('col col--4', styles.blog)}>
                        <div className="text--center">
                            <p>Placeholder for blog</p>
                        </div>
                    </div>
                    <div className={clsx('col col--4', styles.careers)}>
                        <div className="text--center">
                            <p>Placeholder for careers</p>
                        </div>
                    </div>
                    <div className={clsx('col col--4', styles.twitterTimeline)}>
                        <div className="text--center">
                            <TwitterTimeline twitterLink={socialLinks.twitter}/>
                            <div className={styles.twitterText}>
                                <p>What we're saying on Twitter</p>
                                <ArrowLink link={socialLinks.twitter}/>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </section>
    )
}

export default Social;
