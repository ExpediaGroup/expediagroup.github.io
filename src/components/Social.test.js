/*
Copyright 2022 Expedia, Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React from 'react';
import renderer from 'react-test-renderer';
import Social from './Social';

jest.mock('./TwitterTimeline', () => 'TwitterTimeline');
jest.mock('./BlogPosts', () => 'BlogPosts');
jest.mock('./SocialHeader', () => 'SocialHeader');
jest.mock('./SocialFooter', () => 'SocialFooter');
jest.mock('@site/static/posts.json', () => [], { virtual: true });
jest.mock('@site/static/img/careers.jpg', () => 'careers-image-url', { virtual: true });

const SOCIAL_CONFIG = {
    blog: {
        title: 'blog title',
        link: 'https://my/blog',
        footerText: 'blog footer'
    },
    careers: {
        title: 'careers title',
        link: 'https://my/careers',
        footerText: 'careers footer'
    },
    twitter: {
        title: 'twitter title',
        link: 'https://my/twitter',
        footerText: 'twitter footer'
    }
}

it('renders correctly', () => {
    const tree = renderer
        .create(<Social socialConfig={SOCIAL_CONFIG}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
