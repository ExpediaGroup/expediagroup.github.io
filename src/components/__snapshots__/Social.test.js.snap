// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="social"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col col--4"
      >
        <div
          className="socialColumn"
        >
          <div
            className="socialHeaderAndBody"
          >
            <SocialHeader
              title="blog title"
            />
            <div
              className="socialBody"
            >
              <BlogPosts
                posts={Array []}
              />
            </div>
          </div>
          <SocialFooter
            link="https://my/blog"
            text="blog footer"
          />
        </div>
      </div>
      <div
        className="col col--4"
      >
        <div
          className="socialColumn"
        >
          <div
            className="socialHeaderAndBody"
          >
            <SocialHeader
              title="careers title"
            />
            <div
              className="socialBody"
            >
              <a
                href="https://my/careers"
                target="_blank"
              >
                <img
                  alt="careers title"
                  className="careersImage"
                  src="careers-image-url"
                />
              </a>
            </div>
          </div>
          <SocialFooter
            link="https://my/careers"
            text="careers footer"
          />
        </div>
      </div>
      <div
        className="col col--4"
      >
        <div
          className="socialColumn"
        >
          <div
            className="socialHeaderAndBody"
          >
            <SocialHeader
              title="twitter title"
            />
            <div
              className="socialBody"
            >
              <TwitterTimeline
                twitterLink="https://my/twitter"
              />
            </div>
          </div>
          <SocialFooter
            link="https://my/twitter"
            text="twitter footer"
          />
        </div>
      </div>
    </div>
  </div>
</section>
`;
